src/codegen/function.rs

```diff
@@ -438,9 +438,12 @@ pub fn body_chunk_futures(
         let c_par = &analysis.parameters.c_parameters[par.ind_c];
 
         let type_ = env.type_(par.typ);
-        let is_str = matches!(*type_, library::Type::Basic(library::Basic::Utf8));
+        let is_str = matches!(type_, library::Type::Basic(library::Basic::Utf8));
+        let is_slice = matches!(type_, library::Type::CArray(_));
 
-        if *c_par.nullable {
+        if is_slice {
```

Comment:
Would also support a nullable slice while at it :)

Reply:
> AFAIK, nullable annotation on array still produces a &[] (and not a Option<&[]>). So to_vec still applies

That is an issue that should be fixed separately, see https://github.com/gtk-rs/gir/pull/1551

Reply:
However, that PR is still not merged.

If I change my code assuming I can have ` Option<&[]>`, that is:
```rust
if *c_par.nullable {
   ...
} else if (is_slice) {
   ...
} ...
```

I'd end up still having a bug generated code until that PR kicks in.

Isn't it better to change the `if`s order in #1551 where the logic actually changes?

---

(P.S: why having `Option<&[]>`. That's not rust idiomatic. the current behavior (if I am not mistaken): `&[]` even when array is annotated as `nullable` makes sense in the generated rust type

Reply:
Sometimes there is a difference in behaviour between passing `None` and `&[]`. See https://github.com/gtk-rs/gir/issues/1133#issue-882390361 but agree, that can be done in the other PR.